aiida/orm/nodes/data/array/bands/electronic.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+"""Data plugin to represent an electronic band structure."""
+from typing import Optional
+
+from aiida.common.exceptions import ModificationNotAllowed
+from aiida.common.lang import type_check
+from .bands import BandsData
+
+__all__ = ('ElectronicBandsData',)
+
+
+class ElectronicBandsData(BandsData):
+    """Data plugin to represent an electronic band structure.
+
+    It subclasses the ``BandsData`` plugin and adds the ``fermi_energy`` attribute.
+    """
+
+    KEY_FERMI_ENERGY = 'fermi_energy'
```

Comment:
Why have this more complicated `KEY_FERMI_ENERGY` class attribute and then set/get the instance attributes using `set_attribute` and `get_attribute`? Also, on line 35 you then still use `self.fermi_energy`, so why not simply use this everywhere?

Reply:
Not sure what you mean here. The reason for the class attribute is that we need to store the value as an attribute with a certain key. I could have chosen anything, but the same name as the property only seems logical. The attribute is there so as to not have to repeat a string literal. Are you asking why the property `fermi_energy` is there as opposed to letting people use `get_attribute` and `set_attribute` directly?

Reply:
Hmm, I _was_ a little quick in going through the code, so the comment on line 35 didn't really make sense. 😅 I guess my question is: Is it worth adding this extra complexity versus using e.g. just `_fermi_energy` to store the attribute just to avoid a bit of repetition. I.e. why not remove `KEY_FERMI_ENERGY` and set/get the Fermi level by just assigning it to a hidden attribute:

```python
    @property
    def fermi_energy(self) -> Optional[float]:
        """Return the fermi energy or ``None`` if one is not defined."""
        return self._fermi_energy

    @fermi_energy.setter
    def fermi_energy(self, value: float):
        """Set the fermi energy."""
        if self.is_stored:
            raise ModificationNotAllowed('The node is stored and cannot be modified.')

        type_check(value, float)
        self._fermi_energy = value
```

Reply:
How is the value in your example getting persisted to the database though? Note that an attribute of the Python class `ElectronicBandsData` does not automatically get stored in the attributes _column_ of the underlying node in the database. We need to manually do this mapping. The property `fermi_energy` is there just to make life easier for the user. Instead of having to know or figure out which key we use to store the value in the attributes, they can discover it through tab-completion.

Reply:
I see, there was my lack of understanding. I missed the requirement to use these methods to actually store the information in the columns of the database. Thanks for the clarification! 👍